src/_pytest/config/__init__.py

```diff
@@ -915,6 +915,25 @@ def _preparse(self, args, addopts=True):
             else:
                 raise
 
+        if ns.override_ini:
+            for ini_config in ns.override_ini:
```

Comment:
Suggestion: can we please move all the validation code to a separate function, and make that function reuse the work done by `_validate_ini_value` to avoid duplication?

Reply:
Maybe, but I would also rather like to keep this "minimal" / "a no brainer" patch for now.
Can be refactored in the end maybe then.

Reply:
yup, please add a follow-up issue

Reply:
Sure.